### PR TITLE
STYLE: Clang tidy cleanups

### PR DESCRIPTION
**Pull request by hjmjohnson**
More clang-tidy identified cleanups found when building with remote modules.

### CODE REVIEW

File: Modules/IO/Bruker/include/itkBruker2dseqImageIO.h [hunk 1]
```diff
@@ -137,7 +137,7 @@ class ITKIOBruker_EXPORT Bruker2dseqImageIO : public ImageIOBase
 
 private:
   void
-  SwapBytesIfNecessary(void * buffer, SizeValueType numberOfPixels);
```
Comment (dzenanz) [inline Modules/IO/Bruker/include/itkBruker2dseqImageIO.h]: I am not sure whether this rename is helpful.

File: Modules/IO/CSV/include/itkCSVFileReaderBase.h [hunk 2]
```diff
@@ -124,7 +124,7 @@ class ITKIOCSV_EXPORT CSVFileReaderBase : public LightProcessObject
   /** Counts the number of rows and columns in a file and prepares the file
    * for iterative reading using the GetNextField() method. */
   void
-  GetDataDimension(SizeValueType & rows, SizeValueType & columns);
```
Comment (dzenanz) [inline Modules/IO/CSV/include/itkCSVFileReaderBase.h]: This one isn't.

File: Modules/IO/ImageBase/include/itkImageIOBase.h [hunk 3]
```diff
@@ -867,7 +867,7 @@ class ITKIOImageBase_EXPORT ImageIOBase : public LightProcessObject
 
 private:
   bool
-  HasSupportedExtension(const char *, const ArrayOfExtensionsType &, bool tolower = true);
```
Comment (dzenanz) [inline Modules/IO/ImageBase/include/itkImageIOBase.h]: Very helpful.

File: Modules/Numerics/FEM/include/itkFEMLinearSystemWrapperDenseVNL.h [hunk 4]
```diff
@@ -151,24 +151,24 @@ class ITKFEM_EXPORT LinearSystemWrapperDenseVNL : public LinearSystemWrapper
   ScaleSolution(Float scale, unsigned int solutionIndex);
 
   void
-  SwapMatrices(unsigned int matrixIndex1, unsigned int matrixIndex2) override;
+  SwapMatrices(unsigned int MatrixIndex1, unsigned int MatrixIndex2) override;
```
Comment (dzenanz) [inline Modules/Numerics/FEM/include/itkFEMLinearSystemWrapperDenseVNL.h]: The convention for parameter names it to begin with lowercase letter, if I remember correctly. Name synchronization in FEM module should be done the other way.

File: Modules/Core/Common/include/itkThreadSupport.h [hunk 5]
```diff
@@ -52,7 +52,7 @@ using ThreadProcessIdType = pthread_t;
 constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = 0;
 using ITK_THREAD_RETURN_TYPE = void *;
 constexpr ITK_THREAD_RETURN_TYPE ITK_THREAD_RETURN_DEFAULT_VALUE =
-  NULL;                                     /* This is from a c library, and always needs to be NULL, not nullptr */
+  nullptr;                                  /* This is from a c library, and always needs to be NULL, not nullptr */
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkThreadSupport.h]: And these few `NULL`s have a big warning next to them about not changing them. Written by you, no less 😈

File: Modules/Core/Common/test/itkSmartPointerGTest.cxx [hunk 6]
```diff
@@ -129,10 +129,10 @@ TEST(SmartPointer, EmptyAndNull)
   cptr = nullptr;
   EXPECT_TRUE(cptr.IsNull());
 
-  ptr = NULL; // Do not change NULL, null, Null in this file. This file intentionally contains usage of legacy NULL
```
Comment (phcerdan) [inline Modules/Core/Common/test/itkSmartPointerGTest.cxx]: The comment explicitly says to not change NULL here.

**Pull request merged**